src/core/dbus-execute.c

```diff
@@ -1258,6 +1280,60 @@ int bus_exec_context_set_transient_property(
         if (streq(name, "RootImage"))
                 return bus_set_transient_path(u, name, &c->root_image, message, flags, error);
 
+        if (streq(name, "RootHash")) {
+                const void *roothash_decoded;
+                size_t roothash_decoded_size;
+
+                /* To have the same behaviour as when parsing units, ignore this property if there's already a path set */
+                if (!isempty(c->root_hash_path))
+                        return 1;
```

Comment:
nah, as mentioned earlier: this is the code where we parse the parameters for transient units, and in that case we should employ the same logic as when parsing unit files: the later setting overrides the former. Hence, instead of bailing out here, just parse whatever is passed and unset `root_hash_path` in that case, excactly like you do it already further down. hence just drop these two lines without replacement

Reply:
Sorry, I misunderstood and thought by "latter" you meant the RootHashPath, rather than what is set last. Will fix in a second.

Reply:
Fixed and pushed again